packages/consumption/src/modules/requests/itemProcessors/proposeAttribute/ProposeAttributeRequestItemProcessor.ts

```diff
@@ -29,14 +31,24 @@ export class ProposeAttributeRequestItemProcessor extends GenericRequestItemProc
             return attributeValidationResult;
         }
 
+        const proposedAttributeMatchesWithQueryValidationResult = validateAttributeMatchesWithQuery(
+            requestItem.query,
+            requestItem.attribute,
+            CoreAddress.from(""),
+            this.currentIdentityAddress
+        );
+        if (proposedAttributeMatchesWithQueryValidationResult.isError()) {
+            return proposedAttributeMatchesWithQueryValidationResult;
+        }
+
         return ValidationResult.success();
     }
 
     private validateAttribute(attribute: IdentityAttribute | RelationshipAttribute) {
```

Comment:
This method should be static, as it doesn't operate on the data of an instance of the item processor. Same for the other private helpers :-)